src/main/java/org/xbill/DNS/lookup/LookupResult.java

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: BSD-2-Clause
+package org.xbill.DNS.lookup;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.xbill.DNS.Name;
+import org.xbill.DNS.Record;
+
+/** LookupResult instances holds the result of a successful lookup operation. */
+@ToString
+@EqualsAndHashCode
+public final class LookupResult {
+  /** An unmodifiable list of records that this instance wraps, may not be null but can be empty */
+  @Getter private final List<Record> records;
+  /** provides additional detail when records is empty */
+  @Getter private final AdditionalDetail additionalDetail;
+  /** tracks the different CNAME and DNAME aliases for this name */
+  @Getter private List<Name> aliases;
+
+  /**
+   * Construct an instance with the provided records.
+   *
+   * @param records a list of records to return, or null if there was no response
+   * @param additionalDetail additional detail on this response, such as the reason for records
+   *     being null.
+   */
+  public LookupResult(List<Record> records, AdditionalDetail additionalDetail, List<Name> aliases) {
```

Comment:
I'd keep this package-private as long as there's no need to have it public

Reply:
I feel fairly strongly that this should be public because it simplifies testing. If a user of the library want to unit test their code it would be natural to mock a LookupSession. If the test code can not create `LookupResult` instances without resorting to reflection magic (or mocking the actual value object) that adds a lot of complexity to the test setup.

This is something that Apache Kafka suffers from, where the admin api returns value objects without public constructors, and it has been pretty annoying.

Reply:
Well, okay, but then please keep in mind that it becomes public API that cannot be changed anymore.